docs/architecture/2018-06-15-request_enrollment_details.md

```diff
@@ -0,0 +1,46 @@
+# A Title Which Summarizes The Decision
+
+## Authors
+
+- Victor Vrantchan ([@groob](https://github.com/groob))
+
+## Status
+
+Accepted (June 15, 2018)
+
+## Context
+
+When enrolling osquery into a remote server, it's desirable to also have a set of initial fields about the osquery node. Usually, servers like Fleet queue ad-hoc queries to get this information, but it's not always reliable. For example, a osquery host can succeed during the Enroll method, and then immediately crash, leaving the server operator with little context about which osquery host to troubleshoot. Another effect of populating host details via distributed query is that they can add some latency to the enrollment process depending on the query interval. 
+
+## Decision
+
+Add a EnrollmentDetails structure to the RequestEnrollment method which sends the results of a population query as part of the enrollment. Below is a list of chose attributes that launcher will always send:
+
+```
+message EnrollmentDetails {
+    string os_version = 1;
```

Comment:
we tried the map pattern in cloud on the receiving end of the `Kolide::Osquery::Api` gem, for receiving both enrollments and logs

losing the strong typing half way through, and divining out what's in the map was a huge pain point

i'm voting for keeping the proto, even if it is a bit of a pain to lug around

Reply:
i’m ok with it being there being an explicit schema to the proto, but my question was mostly about the growth of the proto over time. I just wasn't sure that fields in the top level is the worth way to manage this lifecycle, which is why I also suggested embedding other typed protos per-topic or something.

I also acknowledge that this is a huge bike shed, so I'm happy to go with whatever really.

Reply:
Every one of these fields is a string. I don't see how a map is any less typed. 
There's value in having the presence of keys in the enrollment details but also a significant cost, as each change requires refactoring multiple places in the data pipeline. 

I'm inclined to follow @marpaia's suggestion of using a map.

Reply:
With an explicit fields I see a situation where some platforms have irrelevant fields, so you end up with them being sent as empty. (imagine some windows or linux specific enrollment details).

Reply:
i would argue that effort adding a key to the map ~= effort adding a key to the proto, since we have to consume on both sides.

i still want a proto, because the structure will be something we can code against. i understand a majority rules here, but we have predictable schema all the way from launcher to this point, and i'd like to keep that if possible

Reply:
I'm going to keep it as a struct in this PR and we can make it a map if it proves to be a bad decision in the future.